lib/pbench/server/auth/__init__.py

```diff
@@ -199,31 +217,33 @@ def get_userinfo(self, token: str = None) -> JSON:
 
         if token:
             self.add_header_param("Authorization", f"Bearer {token}")
-        params_path = {"realm-name": self.realm_name}
 
-        return self._get(URL_USERINFO.format(**params_path)).json()
+        return self._get(self.USERINFO_ENDPOINT).json()
 
-    def logout(self, refresh_token: str) -> HTTPStatus:
+    def logout(self, refresh_token: str):
         """
         The logout endpoint logs out the authenticated user.
         :param refresh_token: Refresh token issued at the time of login
         """
-        params_path = {"realm-name": self.realm_name}
-        payload = {
-            "client_id": self.client_id,
-            "client_secret": self.client_secret_key,
-            "refresh_token": refresh_token,
-        }
+        if self.LOGOUT_ENDPOINT:
+            payload = {
+                "client_id": self.client_id,
+                "client_secret": self.client_secret_key,
+                "refresh_token": refresh_token,
+            }
 
-        return HTTPStatus(
-            self._post(URL_LOGOUT.format(**params_path), data=payload).status_code
-        )
+            return self._post(self.LOGOUT_ENDPOINT, data=payload)
```

Comment:
Where we login above, we add an authorization header with `self.add_header_param`; shouldn't we be removing that here?

Reply:
Does it really matter? I mean yes we can remove the header, but once the token is already revoked it can not be used again. And when we have to get `userinfo` again with a new token we explicitly add it as a header replacing any existing one. If the user has not provided us with new token after they revoked the existing one we propagate 401 back to the user.